bin/templates/cordova/lib/build.js

```diff
@@ -422,11 +500,18 @@ function parseOpts(options, resolvedTarget) {
 module.exports.runClean = function(options) {
     var opts = parseOpts(options);
     var builder = builders[opts.buildMethod];
-    return builder.prepEnv()
+    return builder.prepEnv(opts)
     .then(function() {
-        return builder.clean(opts.extraArgs);
+        return builder.clean(opts);
     }).then(function() {
         shell.rm('-rf', path.join(ROOT, 'out'));
+
+        ['debug', 'release'].forEach(function(config) {
+            var propertiesFilePath = path.join(ROOT, config + SIGNING_PROPERTIES);
+            if(isAutoGenerated(propertiesFilePath)){
+                shell.rm('-f', propertiesFilePath);
```

Comment:
Yes, currently its not wired up to the CLI. We have received feedback that it should be wired. I agree deleting after every build is the most 'secure' - though it breaks the android studio IDE interop story. My thinking is a developer will specify the build.json file with the required parameters and iterate using a combination of Android studio and CLI.
Some of this is hard to predict, I'm of the opinion that let's get it out there and drive some usage and listen to feedback.
Let me know if you feel strongly about removing the deletion of these files from 'clean'. Having this be there 'clean' allows the developer a way to remove these files and not rely on internals.